### ISSUE TITLE
Provide Autostart Evidence dev server setting and implement autostart and install dependencies prompt on vscode load for smoother user onboarding

### ISSUE THREAD

**Issue by RandomFractals** [evidence-dev/evidence-vscode#53]
in addition to new activation events and other checks added in #42 and #43 to:

- [x] Start the Evidence dev server when `evidence.autoStart` configuration setting is set to `true`.
- [x] Set it to `true` by default to autostart Evidence dev server when evidence files are detected on vscode load and extension activation
- [x] When `evidence.autoStart` is set to `true` in the user or workspace configuration settings, and we don't detect the require `node_modules` folder to run dev server, display info prompt and ask if the user wants to install Evidence dependencies.
- [x] Display loading progress bar when server is starting on vscode window load

**Comment by RandomFractals** [comment]
New Autostart settings:

![image](https://github.com/evidence-dev/evidence-vscode/assets/656833/855ea943-ae2c-4fe0-b27d-29a07e5cd94a)

**Comment by RandomFractals** [comment]
Install dependencies prompt when opening new Evidence project from template and it doesn't have `node_modules`:

![vscode-evidence-project-install-dependencies](https://github.com/evidence-dev/evidence-vscode/assets/656833/b4ae65cb-477a-4176-9bd7-c6c77fa7148e)

**Comment by RandomFractals** [comment]
App auto start: (still takes 25 seconds for the page to load 🤷‍♂️ )

![evidence-app-auto-start](https://github.com/evidence-dev/evidence-vscode/assets/656833/44e40540-9d8b-4db6-8718-e89cf5696fb2)

**Comment by RandomFractals** [comment]
Autostart should now default to `true` and start Evidence dev server automatically when our project files are detected on vscode load and ext. activation:

![image](https://github.com/evidence-dev/evidence-vscode/assets/656833/365bd654-1a74-4803-8f53-cb652f6ce7bc)

**Comment by archiewood** [comment]
This does not seem to boot the simple browser successfully

![CleanShot 2023-06-02 at 13 45 23](https://github.com/evidence-dev/evidence-vscode/assets/58074498/d116cbff-217f-45a9-8caf-cd8cb7e856a7)

**Comment by RandomFractals** [comment]
I see the built-in Simpler Browser view open. 

This must be running an older extension version producing those results as that flow is very different now after the changes in this ticket and #56.

**Comment by archiewood** [comment]
Tested at around ~2pm today

**Comment by RandomFractals** [comment]
yeah, that doesn't look right or match the latest code. see: https://github.com/evidence-dev/evidence-vscode/issues/57#issuecomment-1574892475

![evidence-install-dependencies-start-flow](https://github.com/evidence-dev/evidence-vscode/assets/656833/baf1e1ce-037f-49c3-ad3a-5844745ea5eb)

**Comment by archiewood** [comment]
Can we adjust this prompt to just say:
> Would you like to install Evidence?

![CleanShot 2023-06-05 at 12 45 17](https://github.com/evidence-dev/evidence-vscode/assets/58074498/35a2ab3b-067e-49dd-a80f-346c55e433b1)

Most of our new users do not really know what a dev server is, so ideally don't want to introduce this concept too early.

**Comment by archiewood** [comment]
Secondly I have an issue here:

1. Create a new project file
2. Command Palette > Create new project from template
3. Would you like to install dependencies? > Select `Yes`
4. Wait

Expected behaviour: 

Spinner in status bar turns into play button once install is complete

Actual behaviour:

Spinner keeps spinning and I cannot start evidence yet

![CleanShot 2023-06-05 at 12 41 33](https://github.com/evidence-dev/evidence-vscode/assets/58074498/94ba125a-af45-447d-92de-ddd07cece5e9)

**Comment by RandomFractals** [comment]
> Can we adjust this prompt to just say:
> 
> > Would you like to install Evidence?
> 
> ![CleanShot 2023-06-05 at 12 45 17](https://user-images.githubusercontent.com/58074498/243419318-35a2ab3b-067e-49dd-a80f-346c55e433b1.gif)
> 
> Most of our new users do not really know what a dev server is, so ideally don't want to introduce this concept too early.

I am opposed to changing this now. We are installing your app dependencies. Please review and suggest such UI labeling issues when these feature tickets are in the initial creation/dev state. We have too many Install Dependencies labels, notifications messages, and related code to do this UI refactor in Sprint 2.

Feel free to open new UI enhancement request for the following sprints to review this.

Thechinally, I think it's the correct terminology to use and VS Code uses the same terminology too.

**Comment by RandomFractals** [comment]
> Secondly I have an issue here:
> 
> ### Repro
> 1. Create a new project file
> 2. Command Palette > Create new project from template
> 3. Would you like to install dependencies? > Select `Yes`
> 4. Wait
> 
> ### Expected behaviour
> Spinner in status bar turns into play button once install is complete
> 
> ### Actual behaviour
> Spinner keeps spinning and I cannot start evidence yet
> 
> ... 

Also a minor timer delay that we can't really fine tune as mentioned before in other review tickets because we only send a command to the terminal and can only guess or estimate the time it takes to install dependencies. I can't enable start dev server earlier, and eventually that state changes after a few seconds.

You can open a separate UX enhancement ticket for future sprints if you feel strongly about these a few extra seconds timer delays we have no control over with the simple terminal API we've decided to use to save time and let us develop other features in 2 weeks.

Closing this as the core feature implementation is done and should be released in v1.1.0 this week.

**Comment by RandomFractals** [comment]
to find a good balance here I will reduce the delay timer for dependencies installation by 5 seconds.

However, just so you know, all the delay timers and spinner we added in Evidence status bar button for these `npm` commands are superficial, and will very on different platforms and machines. 

They do all expire in a few extra seconds or so. In order to have perfect timings, we need to read running process status and should be switching to using nodejs `child_process` like we do for node version check already. Otherwise, we can't fine-tune them any further.
